client/lib/media-serialization/strategies/dom.js

```diff
@@ -21,7 +21,7 @@ function parseImage( node, _parsed ) {
 	_parsed.type = MediaTypes.IMAGE;
 	_parsed.media.URL = node.getAttribute( 'src' );
 	_parsed.media.alt = node.getAttribute( 'alt' );
-	_parsed.media.transient = ( 0 === ( _parsed.media.URL || '' ).indexOf( 'blob:' ) );
+	_parsed.media.transient = !! node.getAttribute( 'data-istransient' );
```

Comment:
Aside: In unrelated work (https://github.com/WordPress/gutenberg/pull/2374) I encountered some specifications relating to these sorts of attributes:

https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attribute
https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#enumerated-attribute

Since this attribute is custom anyways, it doesn't really matter so much specifically how we treat it. But certainly seems to fall under the criteria of a boolean attribute (i.e. can only be `false` in case that it is not assigned at all).